src/engine.py

```
@@ -9404,31 +9640,19 @@ async def _audiosocket_handle_audio(self, conn_id: str, audio_bytes: bytes) -> N
-                        elif canonical_fmt == "slin16":
-                            profile_rate = 16000
-                        else:
-                            profile_rate = getattr(self.config.streaming, "sample_rate", 8000)
+                    # The TLV message type is the authoritative per-frame format.
+                    profile_fmt = frame_format
+                    profile_rate = frame_rate
                 else:
                     # For RTP: use transport profile (negotiated codec)
                     profile_fmt = session.transport_profile.format or "ulaw"
                     profile_rate = session.transport_profile.sample_rate or 8000
             except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)